engineapi/proxy.go

```diff
@@ -81,9 +84,19 @@ func NewEngineClientProxy(im *longhorn.InstanceManager, logger logrus.FieldLogge
 	}
 
 	ctx, cancel := context.WithCancel(context.Background())
-	client, err := imclient.NewProxyClient(ctx, cancel, im.Status.IP, InstanceManagerProxyServiceDefaultPort)
+	client, err := imclient.NewProxyClientWithTLS(ctx, cancel, im.Status.IP, InstanceManagerProxyServiceDefaultPort,
+		filepath.Join(types.TLSDirectoryInContainer, types.TLSCAFile),
+		filepath.Join(types.TLSDirectoryInContainer, types.TLSCertFile),
+		filepath.Join(types.TLSDirectoryInContainer, types.TLSKeyFile),
+		"longhorn-backend.longhorn-system")
 	if err != nil {
-		return nil, err
+		logrus.WithError(err).Warnf("Falling back to non-tls client for Instance Manager Instance Service Client for %v, state: %v, IP: %v",
```

Comment:
Probably not a good idea to log as warnings, suggest use at the trace level.

Reply:
Any reason we are logging the current state?

Reply:
Removed. Thanks.